File: frontend/app/types/workflow.ts
// 工作流与监控相关类型

export type StepStatus = 'done' | 'running' | 'pending'

export interface PipelineStep {
  name: string
  label: string
  status: StepStatus
  count: string
  progress?: number
}

export interface AgentStatus {
  name: string
  label: string
  status: 'idle' | 'running' | 'pending'
  detail: string
}

export interface LogEntry {
  timestamp: string
  level: 'INFO' | 'WARN' | 'ERROR'
  agent: string
  message: string
}

export interface SystemMetrics {
  cpu: number
  memory: { used: number; total: number }
  redis: number
  dbConnections: number
}

export interface WorkflowState {
  isRunning: boolean
  lastCollectTime: string
  nextCollectTime: string
  totalProgress: number
  estimatedRemaining: string
  pipelineSteps: PipelineStep[]
  agents: AgentStatus[]
  logs: LogEntry[]
  metrics: SystemMetrics
}

File: backend/app/schemas/workflow.py
# 工作流与监控相关 Pydantic Schema — 严格匹配前端 TypeScript 类型
from pydantic import BaseModel, Field
from typing import Optional


class PipelineStep(BaseModel):
    name: str
    label: str
    status: str  # done / running / pending
    count: str
    progress: Optional[int] = None


class AgentStatus(BaseModel):
    name: str
    label: str
    status: str  # idle / running / pending
    detail: str


class LogEntry(BaseModel):
    timestamp: str
    level: str  # INFO / WARN / ERROR
    agent: str
    message: str


class MemoryMetrics(BaseModel):
    used: float
    total: float


class SystemMetrics(BaseModel):
    cpu: float
    memory: MemoryMetrics
    redis: float
    dbConnections: float = Field(alias="dbConnections")

    class Config:
        populate_by_name = True


class WorkflowState(BaseModel):
    isRunning: bool = Field(alias="isRunning")
    lastCollectTime: str = Field(alias="lastCollectTime")
    nextCollectTime: str = Field(alias="nextCollectTime")
    totalProgress: float = Field(alias="totalProgress")
    estimatedRemaining: str = Field(alias="estimatedRemaining")
    pipelineSteps: list[PipelineStep] = Field(default_factory=list, alias="pipelineSteps")
    agents: list[AgentStatus] = []
    logs: list[LogEntry] = []
    metrics: SystemMetrics

    class Config:
        populate_by_name = True
